packages/core/lib/payload-store/JsonStringifyStreamSerializer.ts

```diff
@@ -0,0 +1,27 @@
+import { FsReadableProvider } from '@lokalise/node-core'
+import { JsonStreamStringify } from 'json-stream-stringify'
+import { tmpNameSync } from 'tmp'
+
+import type { PayloadSerializer } from './payloadStoreTypes'
+
+export type TemporaryFilePathResolver = () => string
+const defaultTemporaryFilePathResolver: TemporaryFilePathResolver = () => tmpNameSync()
+
+export class JsonStringifyStreamSerializer implements PayloadSerializer {
+  constructor(
+    private readonly temporaryFilePathResolver: TemporaryFilePathResolver = defaultTemporaryFilePathResolver,
+  ) {}
+
+  async serialize(payload: unknown) {
+    const fsReadableProvider = await FsReadableProvider.persistReadableToFs({
```

Comment:
this is going to leak files. serializer interface should also expose close/cleanup method, which in this case should destroy the provider

Reply:
Addressed in https://github.com/kibertoad/message-queue-toolkit/pull/167/commits/bbe6f6ae8634e5ec65037a7390ed077918b83718